src/main/java/org/opensearch/index/codec/customcodecs/CustomCodecPlugin.java

```diff
@@ -48,6 +48,13 @@ public Optional<CodecServiceFactory> getCustomCodecServiceFactory(final IndexSet
             || codecName.equals(CustomCodecService.QAT_LZ4_CODEC)
             || codecName.equals(CustomCodecService.QAT_DEFLATE_CODEC)) {
             return Optional.of(new CustomCodecServiceFactory());
+        } else {
+            if (!QatZipperFactory.isQatAvailable()
+                && (codecName.equals(Lucene99QatCodec.Mode.QAT_LZ4.getCodec())
```

Comment:
So the Lucene99QatCodec.Mode.QAT_LZ4.getCodec() string value will be `QATLZ499`, whereas the constant defined in CustomCodecService is `qat_lz4`. Do we support defining an index with either codec name?

Reply:
yes, so the codecs are registered with Lucene using ```QATLZ499``` and ```QATDeflate99```. Users can use these values to create indices, and the [validation succeeds](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/engine/EngineConfig.java#L142). Similarly, Lucene99, Zstd99 also works.
With this check, we want to prevent the codec creation with these values.